src/transforms/CountPattern.js

```diff
@@ -19,47 +18,45 @@ function CountPattern(graph) {
   return this.router(true).revises(true);
 }
 
-var prototype = (CountPattern.prototype = Object.create(BatchTransform.prototype));
+var prototype = (CountPattern.prototype = Object.create(Transform.prototype));
 prototype.constructor = CountPattern;
 
-prototype.batchTransform = function(input, data) {
+prototype.transform = function(input) {
   log.debug(input, ['countpattern']);
 
   var get = this.param('field'),
       pattern = this.param('pattern'),
       tcase = this.param('case'),
       stop = this.param('stopwords'),
-      batch = 0;
+      reset = false, run = false;
```

Comment:
You're right. The `Transform` parent class provides a `reset` parameter as part of the `transform` method that is true if any dependent signals re-evaluate. I'll switch over to using + augmenting that, which should take care of this case and simplify CountPattern some more.